queries.sql

```diff
@@ -1,24 +1,94 @@
 SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
-
 --List name and escape attempts of animals that weigh more than 10.5kg.
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
-
 --Find all animals that are neutered.
 SELECT * FROM animals WHERE neutered = true;
-
 --Find all animals not named Gabumon.
 SELECT * FROM animals WHERE name != 'Gabumon';
-
 --Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg).
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+--Inside a transaction update the animals table by setting the species column to unspecified. 
+-- Start a transaction
+BEGIN;
+-- Update the 'species' column to 'unspecified'
+UPDATE animals SET species = 'unspecified';
+-- Verify the change (you can run SELECT statements within the same transaction)
+SELECT * FROM animals WHERE species = 'unspecified';
+-- Roll back the transaction to revert the changes
+ROLLBACK;
+
+--Update the animals table
+-- Start a transaction
+BEGIN;
+-- Update the 'species' column to 'digimon' for animals with names ending in 'mon'
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+-- Update the 'species' column to 'pokemon' for animals with no species set
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+-- Verify the changes (you can run SELECT statements within the same transaction)
+SELECT * FROM animals;
+-- Commit the transaction to save the changes
+COMMIT;
+
+--Inside a transaction delete all records and rollback
+-- Start a transaction
+BEGIN;
+-- Delete all records in the 'animals' table
+DELETE FROM animals;
+-- Roll back the transaction to undo the deletion
+ROLLBACK;
+--  verify if all records in the animals table still exists
+SELECT * FROM animals;
+
+
+-- Create a savepoint for the transaction after deletion.
+-- Start a transaction
+BEGIN;
+-- Delete all animals born after Jan 1st, 2022
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+-- Create a savepoint
+SAVEPOINT my_savepoint;
+-- Update all animals' weight to be their weight multiplied by -1
+UPDATE animals SET weight_kg = weight_kg * -1;
+-- Rollback to the savepoint
+ROLLBACK TO SAVEPOINT my_savepoint;
+-- Update all animals' weights that are negative to be their weight multiplied by -1
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+-- Commit the transaction
+COMMIT;
+
+--Write the queries
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) AS max_escape_attempts
+FROM animals
+GROUP BY neutered;
+
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
```

Comment:
- [ ] As I mentioned previously, after updating data, please include new queries and screenshots, because currently, all screenshot has the wrong output. This will help ensure that your data is in accordance with the project's specifications. Keep up the good work! 👍📊📸